contracts/PriceFeed/PriceFeed.sol

```diff
@@ -101,22 +141,54 @@ contract PriceFeed is IPriceFeed, Initializable, ProxyOwned {
 
     function _getRateAndUpdatedTime(bytes32 currencyKey) internal view returns (RateAndUpdatedTime memory) {
         AggregatorV2V3Interface aggregator = aggregators[currencyKey];
-        require(address(aggregator) != address(0), "No aggregator exists for key");
-
-        // this view from the aggregator is the most gas efficient but it can throw when there's no data,
-        // so let's call it low-level to suppress any reverts
-        bytes memory payload = abi.encodeWithSignature("latestRoundData()");
-        // solhint-disable avoid-low-level-calls
-        (bool success, bytes memory returnData) = address(aggregator).staticcall(payload);
-
-        if (success) {
-            (, int256 answer, , uint256 updatedAt, ) = abi.decode(returnData, (uint80, int256, uint256, uint256, uint80));
-            return
-                RateAndUpdatedTime({rate: uint216(_formatAggregatorAnswer(currencyKey, answer)), time: uint40(updatedAt)});
+        IUniswapV3Pool pool = pools[currencyKey];
+        require(address(aggregator) != address(0) || address(pool) != address(0), "No aggregator or pool exists for key");
+
+        if (aggregator != AggregatorV2V3Interface(0)) {
```

Comment:
I dont get this check, if  `aggregator` is not `Aggregator` then use TWAP?

Reply:
I took a liberty to give a priority to aggregators, so in previous version we were checking here if either aggregator or pool exists for key, if aggregator exists - fetch price from aggregator regardless of the pool.

Since I've added the condition that we cannot add a pool for key if aggregator exists, we won't have duplicates, but I left the logic as it is for now.